### ISSUE TITLE
ci.commands silently ignores non-existent workflow files

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#177]
## Description

When configuring `[process.ci.commands]` with a workflow file that doesn't exist, the check passes silently with no violation reported. This could mask configuration errors (e.g., typos in workflow filenames).

## Steps to Reproduce

1. Create a check.toml with:
```toml
[process.ci]
enabled = true

[process.ci.commands]
"non-existent-workflow.yml" = ["npm test"]
```

2. Run the check:
```bash
npx check-my-toolkit process check -f json
```

3. **Expected**: Violation reported for non-existent workflow file
4. **Actual**: Check passes with no violations

## Root Cause

In `src/process/tools/ci.ts`, the `checkWorkflowCommands` method:

```javascript
const { workflow, parseError } = this.parseWorkflow(projectRoot, wf);
if (parseError) {
    violations.push(this.yamlErrorViolation(wf, parseError));
    continue;
}
if (!workflow) {
    continue;  // <-- SILENTLY SKIPS non-existent files
}
```

When `parseWorkflow` reads a non-existent file, it returns `{ workflow: null }` without a `parseError`. The code then silently continues to the next workflow.

## Suggested Fix

Add a violation when workflow file doesn't exist:

```javascript
if (!workflow) {
    violations.push({
        rule: `${this.rule}.commands`,
        tool: this.toolId,
        file: `.github/workflows/${wf}`,
        message: `Workflow file '${wf}' not found`,
        severity: "error",
    });
    continue;
}
```

## Version

1.6.0
